main/src/main/scala/sbt/ScriptedPlugin.scala

```diff
@@ -188,10 +190,22 @@ object ScriptedPlugin extends AutoPlugin {
       val opts = scriptedLaunchOpts.value.toArray
       val empty = new java.util.ArrayList[File]()
       val instances = Int box scriptedParallelInstances.value
+      val reuseServer = Boolean box scriptedReuseServer.value
 
       if (scriptedBatchExecution.value)
-        method.invoke(scriptedInstance, dir, log, args.toArray, launcher, opts, empty, instances)
-      else method.invoke(scriptedInstance, dir, log, args.toArray, launcher, opts, empty)
```

Comment:
Doesn't the semantics of `scriptedBatchExecution.value` already mean that we're reusing the server?
The only new thing is that batch execution became available to sbt plugin authors?

Reply:
I could be mistaken, but it appeared in my testing that the semantics of scriptedBatchExecution seem to be that it will spin up a bunch of sbt servers in parallel and run all the scripted tests. This is very bad because they all contend with each other for the ivy lock.

Reply:
If I understand correctly, the "batching" was named so because it grouped scripted tests into some batches, and run them together, AND those batches run in parallel. Whereas traditional scripted test would spin up a new sbt server with a cold JVM per test sequentially. Thus, the idea of reusing occurs only during batching.

Reply:
Got it. So somehow during refactoring, that behavior was changed and batch execution became the default with no parallelism. This is a really nice improvement for most use cases except where there are problems clearing the workspace like in the paradox use case. It seems that the real issue then is to get scripted to honor the scriptedBatchExecution flag.